wire-schema/src/main/java/com/squareup/wire/schema/Location.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema
+
+import com.google.common.base.CharMatcher
+import java.io.File
+
+/**
+ * Locates a .proto file, or a position within a .proto file, on the file system. This includes a
+ * base directory or a .jar file, and a path relative to that base.
+ */
+data class Location(
+  /** The base of this location; typically a directory or .jar file.  */
+  val base: String,
+
+  /** The path to this location relative to [.base].  */
+  val path: String,
+
+  /** The line number of this location, or -1 for no specific line number.  */
+  val line: Int = -1,
+
+  /** The column on the line of this location, or -1 for no specific column.  */
+  val column: Int = -1
+) {
+
+  fun at(
```

Comment:
nit: I'd prefer this to be a one-liner

Reply:
reverted, although the formatter sets it back :(

Reply:
fixed